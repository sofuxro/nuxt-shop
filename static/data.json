[
    {
        "id": "7e3623a7-7bf8-4855-9819-f4d82af9e1c5",
        "name": "Desi Avramovitz",
        "price": 5000.4,
        "image": "audemars-01-1-190x206@2x.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "cc3adec3-7f10-4eef-afe2-80376cfe1357",
        "name": "Arden Stockbridge",
        "price": 10000.0,
        "image": "seamaster-sp-01-190x206@2x.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "544ac97b-419a-472a-8edc-f42a55324da0",
        "name": "Addy Alldre",
        "price": 3300.99,
        "image": "audemars-02-350x380.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "6691bbf5-5ade-446f-89cf-c1a664d58d24",
        "name": "Bernie Gledhill",
        "price": 10200.99,
        "image": "globemaster-01-190x206@2x.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "885e1249-c45d-4c49-baa6-6434a291d4d2",
        "name": "Gwendolen Bickerstaffe",
        "price": 6400.5,
        "image": "globemaster-02-350x380.jpg",
        "tags": { "gender": "male", "featured": true }
    },
    {
        "id": "6df0a393-2d01-4024-a46d-1d06b200458a",
        "name": "Cristian Gilbanks",
        "price": 3000.99,
        "image": "marina-01-190x206@2x.jpg",
        "tags": { "gender": "male", "featured": true }
    },
    {
        "id": "3223686e-a223-472d-8dd1-6b944aadc222",
        "name": "Kalila Gooms",
        "price": 1900.99,
        "image": "marina-02-350x380.jpg",
        "tags": { "gender": "female" }
    },
    {
        "id": "ac3470c1-a0ea-4712-8cd5-d61ac8d30643",
        "name": "Bartolemo Peckitt",
        "price": 13000.0,
        "image": "montbrilliant-01-190x206@2x.jpg",
        "tags": { "gender": "male", "featured": true }
    },
    {
        "id": "00c8b285-a944-4386-a637-0f32514246df",
        "name": "Kathye Haith",
        "price": 12000.99,
        "image": "montbrilliant-02-350x380.jpg",
        "tags": { "gender": "male", "featured": true }
    },
    {
        "id": "70e337e6-811a-45b3-8515-ef4f5c21869f",
        "name": "Armin Basilio",
        "price": 2000.99,
        "image": "moto360-01-190x206@2x.jpg",
        "tags": { "gender": "female", "featured": true }
    },
    {
        "id": "9dcd7387-4162-4a2c-b2cd-67bd3a3e5a2f",
        "name": "Arden Stockbridge",
        "price": 10000.0,
        "image": "seamaster-sp-02-350x380.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "421900ad-ecb7-4a0b-b80c-3ccc877a0dca",
        "name": "Cal Sterman",
        "price": 5000.0,
        "image": "omega-speed-01-1-190x206@2x.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "ec97cff1-ac0a-487c-ab0d-ffb240fc9128",
        "name": "Ardine Imlaw",
        "price": 6500.0,
        "image": "omega-speed-02-350x380.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "6488a504-5825-4a46-b6d5-546254d56ce5",
        "name": "Noella Ruddom",
        "price": 7200.0,
        "image": "patrimony-01-190x206@2x.jpg",
        "tags": { "gender": "female" }
    },
    {
        "id": "ce06bf75-3c50-4082-bb18-4a11f4b816fc",
        "name": "Arden Stockbridge",
        "price": 10000.0,
        "image": "tag-carrer-01-190x206@2x.jpg",
        "tags": { "gender": "female" }
    },
    {
        "id": "57b1c56c-d1be-44b6-857d-4dec059d3fde",
        "name": "Kirstyn Espadate",
        "price": 5900.99,
        "image": "patrimony-02-350x380.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "daef0fa6-7fde-49f3-9788-a3fb2f328960",
        "name": "Rourke Greet",
        "price": 4900.99,
        "image": "radomir-01-190x206@2x.jpg",
        "tags": { "gender": "male" }
    },
    {
        "id": "6be78686-6769-46d7-9abc-d5fe74960292",
        "name": "Arden Stockbridge",
        "price": 10000.0,
        "image": "radomir-02-350x380.jpg",
        "tags": { "gender": "female" }
    },
    {
        "id": "f9c88546-8725-4038-bc06-fb9e27e9033b",
        "name": "Arden Stockbridge",
        "price": 10000.0,
        "image": "tag-carrer-02-350x380.jpg",
        "tags": { "gender": "female" }
    }
]
