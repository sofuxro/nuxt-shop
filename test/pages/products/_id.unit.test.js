import { factoryMount } from '~/test/utils/factories'
import flushPromises from 'flush-promises'

import Component from '~/pages/products/_id'
import data from '~/static/data.json'

let wrapper
const mock = jest.fn()

beforeEach(() => {
    wrapper = factoryMount(Component, {
        store: {
            modules: {
                products: {
                    namespaced: true,

                    getters: {
                        productById: (state) => (id) => data[0]
                    },

                    actions: {
                        addToCart: mock
                    }
                }
            }
        },
        mocks: {
            $route: { params: '' }
        },
        stubs: ['Featured']
    })
})

afterEach(() => {
    wrapper.destroy()
})

describe('Product page', () => {
    test('is fully functional', async () => {
        const btnAdd = wrapper.find('button')
        btnAdd.trigger('click')
        await flushPromises()
        expect(mock).toHaveBeenCalledTimes(1)

        // Snapshot
        expect(wrapper.element).toMatchSnapshot()
    })
})
