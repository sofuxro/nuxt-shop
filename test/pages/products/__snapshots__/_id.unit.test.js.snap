// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page is fully functional 1`] = `
<div
  class="container watch-page"
>
  <div
    class="item"
  >
    <div
      class="v-responsive v-image item__image"
    >
      <div
        class="v-responsive__content"
      />
    </div>
     
    <div
      class="item__content"
    >
      <div
        class="display-1"
      >
        Desi Avramovitz
      </div>
       
      <div
        class="price headline font-weight-thin py-4"
      >
        
                $5,000.40
            
      </div>
       
      <div
        class="font-weight-thin"
      >
        
                Lorem ipsum dolor sit, amet consectetur adipisicing elit.
                Asperiores voluptatibus illum maiores illo quod repellat
                dolores, sapiente pariatur, assumenda vitae cum aperiam
                excepturi ullam modi alias magnam animi hic quia blanditiis
                id saepe, rem nobis. Fugiat corrupti quae obcaecati
                inventore repudiandae facilis tenetur voluptate, quisquam ex
                temporibus illo praesentium distinctio provident vel ducimus
                eius libero veniam exercitationem. Vero impedit
                necessitatibus magnam perferendis! Dolorum culpa explicabo
                aliquam praesentium rem, ipsum repellat ea aspernatur
                obcaecati consequuntur suscipit odio autem sed libero animi
                tempora odit iusto ab, in assumenda totam vitae a,
                blanditiis sapiente. Voluptatem voluptates enim commodi
                soluta corporis iure labore ipsa.
                
        <br />
         
        <b>
          This is a Demo Online Store. No orders shall be
                    fulfilled.
        </b>
      </div>
       
      <button
        class="mt-6 v-btn v-btn--depressed theme--light v-size--default orange lighten-1"
        style="min-width: 240px;"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          ADD TO CART
        </span>
      </button>
    </div>
  </div>
   
  <featured-stub />
</div>
`;
